discovery-provider/src/tasks/tracks.py

```diff
@@ -172,6 +173,102 @@ def update_remixes_table(session, track_record, track_metadata):
                     )
                     session.add(remix)
 
+
+def update_track_routes_table(session, track_record, track_metadata):
+    # Check if the title is staying the same, and if so, return early
+    if track_record.title == track_metadata['title']:
+        return
+
+    # Get the title slug, and set the new slug to that
+    # (will check for conflicts later)
+    new_track_slug_title = (
+        helpers.create_track_slug(track_metadata['title'])
+    )
+    new_track_slug = new_track_slug_title
+
+    # Find the current route for the track
+    prev_track_route_record = (
+        session
+        .query(TrackRoute)
+        .filter(TrackRoute.track_id == track_record.track_id,
+                TrackRoute.is_current == True)  # noqa: E712
+        .one_or_none()
+    )
+
+    if prev_track_route_record is not None:
+        if prev_track_route_record.title_slug == new_track_slug_title:
+            # If the title slug hasn't changed, we have no work to do
+            return
+        # The new route will be current
+        prev_track_route_record.is_current = False
+
+    # Check for conflicts Find the max conflict ID for the track slug
+    max_collision_id = (
+        session
+        .query(functions.max(TrackRoute.collision_id))
+        .filter(TrackRoute.title_slug == new_track_slug_title,
+                TrackRoute.owner_id == track_record.owner_id)
+        .one_or_none()
+    )[0]
```

Comment:
I like this one_or_none! Can safely index into the list, nice. Gonna use this one going forward